src/libs/sfdk/buildengine.h

```diff
@@ -124,6 +124,10 @@ class SFDK_EXPORT BuildEngine : public QObject
     void importPrivateGpgKey(const QString &id, const Utils::FilePath &passphraseFile,
             const QObject *context, const Functor<bool, QString> &functor);
 
+    int freeSize() const;
+    void setFreeSizeMb(int freeSizeMb, int incrementMb, const QObject *context,
+            const Functor<bool> &functor);
+
```

Comment:
Compare this with `setStorageSizeMb`, which is also not supported on emulator. Still it is exposed on `VirtualMachine` and we have those `GrowStorageSize` and `ShrinkStorageSize` feature flags to indicate its availability. Similarly we should expose `freeStorageSizeMb` and `reserveStorageSizeMb` (*) on `VirtualMachine` and add the feature flag `ReserveStorageSize` to indicate their availability.

(*) Before I suggested to use "ensure" in the name, but "reserve" will work better for the feature flag, so let's use that instead.

Reply:
The reason I moved it to `BuildEngine` is that `VirtualMachine` does no consept of shared folders - those are implemented on `BuildEngine` and `Emulator` level. Are you sure you want me to move the shared folders to `VirtualMachine`? Well, `sharedConfigPath` at least. Both emulators and build engines have those, so it would make kind of sense.

Reply:
Actually the information is available at that level, even that it is not exposed in the public interface of `VirtualMachine` - you can get the shared path as `d->cachedInfo().sharedConfig` there. See how it is used in `DockerVirtualMachinePrivate::makeCreateArguments()`.

You will need to add the `ReserveStorageSize` flag to `VBoxVirtualMachine::staticFeatures` and to `EmulatorPrivate::initVirtualMachine()::unsupportedFeatures` to indicate its availability just on vbox-based build engine.

Reply:
Thanks! I do have to wonder how I missed that.